### PR TITLE
Added to plugin optional opportunity generation Play client.

### PR DESCRIPTION
**Pull request by atyutin90**
There was added opportunity to exclude generation scala file with injection for Play.

### CODE REVIEW

File: plugin/src/main/java/play/soap/plugin/Configuration.java [hunk 1]
```diff
@@ -44,4 +48,25 @@ public static boolean exists(String name) {
...
+
+    final String name;
+
+    Target(String name) {
+      this.name = name;
+    }
+
+    public static Target findByName(String name) {
+      return Arrays.stream(values())
+          .filter(lang -> lang.name.equals(name))
```
Comment (ihostage) [inline plugin/src/main/java/play/soap/plugin/Configuration.java:63]: I think we can compare values without case sensitive. The same behaviour can be for `LANG` parameter.

```suggestion
          .filter(target -> target.name.equalsIgnoreCase(name))
```

File: docs/modules/ROOT/pages/plugin/cli.adoc [hunk 2]
```diff
@@ -4,5 +4,5 @@ Use https://cxf.apache.org/docs/wsdl-to-java.html[wsdl2java] with options:
 [,cmd]
 ----
 CLASSPATH=../../play-soap-plugin.jar
-wsdl2java -fe play -xjc-Xplay:langjava helloWorld.wsdl
+wsdl2java -fe play -xjc-Xplay:langjava -xjc-Xplay:targetplay helloWorld.wsdl
```
Comment (ihostage) [inline docs/modules/ROOT/pages/plugin/cli.adoc:7]:
```suggestion
wsdl2java -fe play -xjc-Xplay:langjava -xjc-Xplay:target play helloWorld.wsdl
```

File: plugin/src/main/java/play/soap/plugin/Configuration.java [hunk 3]
```diff
@@ -9,10 +9,14 @@
 
 public class Configuration {
 
+  public static String XJC = "xjc";
   public static String OPTION_PREFIX = "Xplay";
 
   public enum Option {
-    LANG("lang");
+    LANG("lang"),
+
```
Comment (ihostage) [inline plugin/src/main/java/play/soap/plugin/Configuration.java:17]: I think this empty line isn't needed 😉

File: plugin/src/main/java/play/soap/plugin/PlayXjcPlugin.java [hunk 4]
```diff
@@ -50,6 +52,13 @@ private int parseLang(String arg) {
         throw new IllegalArgumentException("Unknown lang: \"" + value + "\"");
       }
     }
+    if (arg.startsWith(TARGET.name)) {
```
Comment (ihostage) [inline plugin/src/main/java/play/soap/plugin/PlayXjcPlugin.java:55]: I think the name of this function `parseLang` should be change.

File: docs/modules/ROOT/pages/plugin/how-to-use.adoc [hunk 5]
```diff
@@ -17,6 +17,10 @@ Start by adding the dependency `play-soap-plugin` to the project and set require
 `scala` - `scala.concurrent.Future` for Scala projects.
 
 `java` - `java.util.concurrent.CompletionStage` for Java projects.
+|-xjc-Xplay:target
+|play |false |Generate the client for SOAP service.
+
+`play` - client for Play
```
Comment (ihostage) [inline docs/modules/ROOT/pages/plugin/how-to-use.adoc:23]:
```suggestion
|-xjc-Xplay:target
|play |false |Generate the additional useful classes to using SOAP service with specified framework.

`play` - Play Framework
```

Comment (ihostage) [reply docs/modules/ROOT/pages/plugin/how-to-use.adoc:23]: @BillyAutrey Hi, Billy! 👋 Can you help me with this formulation? Is my suggestion ok?

**Review by ihostage — CHANGES_REQUESTED** [review]
Thanks, Andrey! 👍 
A few small comments are below 😉

File: docs/modules/ROOT/pages/plugin/how-to-use.adoc (hunk 5 above)
Comment (ihostage) [reply docs/modules/ROOT/pages/plugin/how-to-use.adoc:23]: Hi, @BillyAutrey!

> Is the idea here that specifying the `play` parameter will generate Play Framework client code?

Yes, you are right.

> Is the code truly a client, or a service?

The name of this class has postfix `Service`. But this class helps to developer create a client for external SOAP service 😄 For example:

```scala
@javax.inject.Singleton
class PrimitivesService @javax.inject.Inject() (apacheCxfBus: play.soap.ApacheCxfBus, configuration: play.api.Configuration) extends play.soap.PlaySoapClient(apacheCxfBus, configuration) {


  lazy val primitives: Primitives = {
    createPort[Primitives](new QName("http://testservice.soap.play/primitives"), "Primitives", "http://0.0.0.0:8080/primitives")
  }

  def primitives(handlers: Handler[_ <: MessageContext]*): Primitives = {
    createPort[Primitives](new QName("http://testservice.soap.play/primitives"), "Primitives", "http://0.0.0.0:8080/primitives", handlers: _*)
  }

  @annotation.varargs
  def getPrimitives(handlers: Handler[_ <: MessageContext]*): Primitives = primitives(handlers: _*)
}
```

> Is there intent to extend this for additional frameworks/clients?

Yes, my long-term plan has this extending. That why I requested @atyutin90 to add special parameter for that. At least, if someone wants to use Play SOAP library without Play Framework, he will can disable generation of classes that has an import of Play Framework classes.

File: docs/modules/ROOT/pages/plugin/how-to-use.adoc [hunk 6]
```diff
@@ -17,6 +17,10 @@ Start by adding the dependency `play-soap-plugin` to the project and set require
 `scala` - `scala.concurrent.Future` for Scala projects.
 
 `java` - `java.util.concurrent.CompletionStage` for Java projects.
+|-xjc-Xplay:target
+|play |false |Generate the additional useful classes to using SOAP service with specified framework.
```
Comment (BillyAutrey) [inline docs/modules/ROOT/pages/plugin/how-to-use.adoc:21]:
```suggestion
|play |none |Generates SOAP client classes for the specified framework.
```

File: docs/modules/ROOT/pages/plugin/how-to-use.adoc (hunk 5 above)
Comment (ihostage) [reply docs/modules/ROOT/pages/plugin/how-to-use.adoc]: > The options are also a little weird here. For Scala readability, I might make the options for `target` either `play` or `none`.

If you about `false`, it isn't a value of `target` param, it's a value of _Required_ column.

File: docs/modules/ROOT/pages/plugin/how-to-use.adoc (hunk 6 above)
Comment (ihostage) [reply docs/modules/ROOT/pages/plugin/how-to-use.adoc]: I changed the text in the last column by your suggestion. But I kept `false` value for `Required` column. WDYT,  @BillyAutrey?

Comment (BillyAutrey) [reply docs/modules/ROOT/pages/plugin/how-to-use.adoc]: LGTM!

**Review by ihostage — APPROVED** [review]

**Pull request merged**